### PR TITLE
keep the array order of includes

### PR DESCRIPTION
**Pull request by steverweber**
keep the array order of includes
https://github.com/saltstack/salt/issues/47954

Tests written?: No

### CODE REVIEW

File: salt/pillar/__init__.py [hunk 1]
```diff
@@ -761,7 +761,7 @@ def render_pstate(self, sls, saltenv, mods, defaults=None):
                                      '\'{0}\' found'.format(saltenv)]
                                 )
 
-                        for sub_sls in set(matched_pstates):
+                        for sub_sls in matched_pstates:
```
Comment (gtmanfred) [inline salt/pillar/__init__.py]: These will still need to be uniqued, because it is possible for multiple states to include the same state.

**Review by gtmanfred — CHANGES_REQUESTED** [review]

**Review by isbm — CHANGES_REQUESTED** [review]

**Comment by cachedout** [comment]
We will want @RichardW42 to look at this since he made the original change. This seems like it should be safe but I would like to hear what he thinks.

File: salt/pillar/__init__.py (hunk 1 above)
Comment (RichardW42) [reply salt/pillar/__init__.py]: @isbm 
```
[_matched_pstates.append(el) for el in matched_pstates if el not in _matched_pstates]
```

Will keep the first item of duplications, but I think in isse #47954 , @steverweber expected keep the last one of duplications.

**Comment by RichardW42** [comment]
@cachedout It seems safe to me. But the concerns about unique includes still needs discussion. If there are multiple states includes:

- A:no unique function applied
  -  A1: pillar_includes_override_sls set to false(which is defaut) or not set: those states will be executed multiple times in order.
  - A2: pillar_includes_override_sls set to true, states will be merged and executed.
- B: unique function applied
  - B1: keep the first item of duplications
  - B2: keep the last item of duplications

As for now, I prefer A1, which is what this mr does.

File: salt/pillar/__init__.py (hunk 1 above)
Comment (RichardW42) [reply salt/pillar/__init__.py]: I noticed that. Maybe I am over-thinking about this case:

```
>>> matched_pstates = ['a', 'b', 'a']
>>> _matched_pstates = []
>>> [_matched_pstates.append(el) for el in matched_pstates if el not in _matched_pstates]
>>> _matched_pstates
['a', 'b']
>>>
```
Which comes to a my consideration that this `a` state should be keep before `b` or after?

Comment (RichardW42) [reply salt/pillar/__init__.py]: Yes, I was thinking about overring/merge policy. Later of this block of code theres was an option called "pillar_includes_override_sls", maybe this unique policy should also provided as an option with default value set, so users can choice to use unique or override or merge policy.

**Comment by rallytime** [comment]
@steverweber Did you get a chance to review this feedback? Where are we on this PR?

**Comment by steverweber** [comment]
yes i seen the feedback... I did do some thought before removing the `set()` yes duplication are not handled but they never prevented in the past releases.

removing duplicates will change the behaviour that was not in the previous releases. /So some caution/.

Personally I would merge this now and perhaps add removing duplication as a new PR. Feedback from @RichardW42 would be nice... lets look at it like this.  If someone had caught that the change from Richard would change the ordering of the includes it likely would not have been merged.

**Comment by steverweber** [comment]
also duplicates might be important because i think salt allows something like
```
include:
   - .add_user
        defaults:
            user: steve
   - .add_user 
        defaults:
            user: joe
```

**Comment by gtmanfred** [comment]
That makes sense.

**Review by gtmanfred — APPROVED** [review]

**Review by terminalmage — APPROVED** [review]
Looks good.

**Pull request merged**